apps/web-giddh/src/app/settings/integration/institutions-list/institutions-list.component.ts

```diff
@@ -0,0 +1,152 @@
+import { Component, OnInit, OnDestroy, ChangeDetectionStrategy, ChangeDetectorRef, Inject } from '@angular/core';
+import { debounceTime, takeUntil } from 'rxjs/operators';
+import { Observable, ReplaySubject } from 'rxjs';
+import { InsitutionsRequest } from '../../../models/api-models/SettingsIntegraion';
+import { SettingIntegrationComponentStore } from '../utility/setting.integration.store';
+import { FormBuilder, FormGroup } from '@angular/forms';
+import { GeneralService } from '../../../services/general.service';
+import { MAT_DIALOG_DATA, MatDialog } from '@angular/material/dialog';
+
+@Component({
+    selector: 'institutions-list',
+    styleUrls: ['./institutions-list.component.scss'],
+    templateUrl: './institutions-list.component.html',
+    providers: [SettingIntegrationComponentStore],
+    changeDetection: ChangeDetectionStrategy.OnPush
```

Comment:
Why use thi

Reply:
due to not getting response after any action